internal/store/device.go

```diff
@@ -659,18 +666,25 @@ func (s *DeviceStore) updateRendered(ctx context.Context, orgId uuid.UUID, name,
 
 	err = ErrorFromGormError(result.Error)
 	if err != nil {
-		return strings.Contains(err.Error(), "deadlock"), err
+		return strings.Contains(err.Error(), "deadlock"), "", err
 	}
 	if result.RowsAffected == 0 {
-		return true, flterrors.ErrNoRowsUpdated
+		return true, "", flterrors.ErrNoRowsUpdated
 	}
-	return false, nil
+	return false, nextRenderedVersion, nil
 }
 
-func (s *DeviceStore) UpdateRendered(ctx context.Context, orgId uuid.UUID, name, renderedConfig, renderedApplications string) error {
-	return retryUpdate(func() (bool, error) {
-		return s.updateRendered(ctx, orgId, name, renderedConfig, renderedApplications)
-	})
+func (s *DeviceStore) UpdateRendered(ctx context.Context, orgId uuid.UUID, name, renderedConfig, renderedApplications string) (string, error) {
+	var (
+		retry bool
+		rv    string
+		err   error
+	)
+	i := 0
```

Comment:
what about something like:
```go
func (s *DeviceStore) UpdateRendered(ctx context.Context, orgId uuid.UUID, name, renderedConfig, renderedApplications string) (string, error)
    var (
		rv    string
	)

	wrapper := func() (bool, error) {
		var retry bool
        var err error
		retry, rv, err = s.updateRendered(ctx, orgId, name, renderedConfig, renderedApplications)
		return retry, err
	}

	err  := retryUpdate(wrapper)
	return rv, err
}
```

Obviously up to you, but this certainly is a for loop